pages/api/auth/magic.ts

```diff
@@ -0,0 +1,36 @@
+import { HttpMethod } from '@lib/http-method';
+import { sendMagicLink } from '@lib/services/magic';
+import { StatusCodes } from 'http-status-codes';
+import { NextApiRequest, NextApiResponse } from 'next';
+
+const MCMASTER_EMAIL_SUFFIX = '@mcmaster.ca';
+
+export default function handler(req: NextApiRequest, res: NextApiResponse) {
+  switch (req.method as HttpMethod) {
+    case HttpMethod.POST:
+      return sendMagicLinkHandler(req, res);
+    default:
+      return res.status(StatusCodes.INTERNAL_SERVER_ERROR).end();
```

Comment:
Would something like 400 Bad Request make more sense here?

Reply:
This should be a [405 Method Not Allowed](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405), thanks